### PR TITLE
Change mean to sum in mixture likelihood

### PR DESCRIPTION
**Pull request by j-towns**
I believe that the log-likelihood of the mixture should be calculated by summing the log-likelihoods of each component, rather than taking their mean. This change may affect how soon EM converges, since the magnitude of updates will (typically) increase relative to tolfun (default 10e-6). It will also affect output when `options.verbosity >= 2`.

### CODE REVIEW

**Comment by j-towns** [comment]
MATLAB `fitgmdist` use the actual log likelihood (which caused me some confusion when comparing to your implementation) whereas others (e.g. [scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/c957249/sklearn/mixture/gmm.py#L524)) appear to do things the same way as you. Note though, that dividing by the number of data (as you do), will lead to _smaller_ changes in the log-likelihood, which will lead to _faster_ triggering of the stopping criterion (i.e. _fewer_ steps of EM). Technically, my change would mean that accuracy _increases_ with the number of data.

**Comment by j-towns** [comment]
I've dug a bit deeper and I was wrong: the comparison which triggers stopping in MATLAB's EM implementation is actually `llDiff < options.TolFun *abs(ll)`(line 276 in MATLAB_R2015b.app/toolbox/stats/stats/@gmdistribution/private/gmcluster.m).
The criteria used will, of course, make a significant difference to the time taken to converge, which you use in your [paper](http://arxiv.org/abs/1506.07677) to measure algorithm performance.

**Comment by reshadh** [comment]
That is a good observation! I actually had seen the MATLAB's stopping criterion and I think it is not that meaningful. Average log-likelihood value (ll) can be arbitrary large or small and its absolute value does not say anything about goodness of it. Consider that all data is multiplied by a constant, then ll value is added by logarithm of that constant. By using MATLAB's stopping criterion, if the data is multiplied just by a constant, the algorithm can stop arbitrarily very early or very late. ll difference is, however, scale invariant and multiplying data by a number does not change ll difference. Therefore, either ll difference, norm of parameter difference or norm of gradient is used for stopping criterion in statistics.
